### PR TITLE
mgr/devicehealth: use helpers to define options and commands

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
i have a patchset in my local repo to use the annotations for

- rebuilding the argdesc string
- collecting the kwargs passed to handlers.

but i want to get this changeset reviewed first.

**Comment by tchaikov** [comment]
jenkins retest this please

**Comment by tchaikov** [comment]
@yaarith i am adding you as a reviewer also in case you are interested in health monitoring.

**Comment by yaarith** [comment]
thanks @tchaikov!
Is this a part of a general change to the mgr modules?
Do we intend to backport these changes?

**Comment by tchaikov** [comment]
@yaarith yes, i am trying rework the way how we generate the command descriptions, so they can be generated from the command's handler function signature automatically.

no, i don't plan to backport this change or the follow-up change of #38722. the whole idea is to improve the maintainability of commands.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-01-06_02:57:51-rados-wip-kefu-testing-2021-01-05-2058-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-01-07_03:01:15-rados-wip-kefu-testing-2021-01-05-2058-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-01-08_08:11:44-rados-wip-kefu-testing-2021-01-08-1210-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-01-10_03:55:58-rados-wip-kefu-testing-2021-01-08-1210-distro-basic-smithi/

failures are tracked by

- https://tracker.ceph.com/issues/48789
- https://tracker.ceph.com/issues/48819

**Comment by tchaikov** [comment]
@epuertat @yaarith could you please help review this changeset?

File: src/pybind/mgr/mgr_module.py [hunk 1]
```diff
@@ -309,6 +311,8 @@ def _parse_args(self):
 
     def __call__(self, func):
         self.func = func
+        if not self.desc:
+            self.desc = inspect.getdoc(func)
```
Comment (epuertat) [inline src/pybind/mgr/mgr_module.py:315]: Nice improvement! We are close to infer the command name from the function (__ => whitespace, _ => -) and the argument types from type annotations:
```py
@CLICommandRead
def device__predict_life_expectancy(devid: CephString):
...
```

**Review by epuertat — APPROVED** [review]
Nice job @tchaikov !

**Pull request merged**